4-generalization/22-bootstrapping/notes.qmd

```diff
@@ -0,0 +1,261 @@
+
+-   When the sample size is large relative to the size of the population, we "blow up" our sample to be the same size as the population. For example if we sample 100 tickets from a box of 200, then we create a bootstrap population by putting two tickets in the box for every ticket in our sample. Then we take a SRS from this bootstrap population. We won't be considering this approach further in these notes.
+
+Let's create the bootstrap sampling distribution for the example of food safety scores from the earlier notes on confidence intervals. In this example, we have the population of food safety scores from all restaurants in San Francisco, and to demonstrate the ideas in the triptych, we took a simple random sample of 100 of these scores. Now, when bootstrapping, that sample becomes our bootstrap population.
+
+```{r}
+#| warning: false
+#| fig-width: 5
+#| fig-height: 3
+
+load(url("https://www.dropbox.com/s/nbgw1uzj5ccwce2/fs_scores.rda?dl=1"))
+
+set.seed(103122)
+samp_scores <- sample(fs_scores, 100)
+
+# Create the bootstrap population
+
+boot_pop <- samp_scores
+
+boot_pop_dist <- ggplot(as.data.frame(boot_pop), aes(x=boot_pop, y=..density..)) + 
+      geom_histogram(breaks=seq(47.5, 100.5, by = 1), 
+                 color="black", fill="gray") + 
+      xlab("Food Safety Scores") +
+      ylab("Density") +
+      ggtitle("Bootstrap Population")
+
+boot_pop_dist
+```
+
+Next, to approximate the bootstrap sampling distribution, we proceed with a simulation. We take a sample of 100, this time with replacement, from the bootstrap population, and we compute the mean food safety score. This process gets repeated 10,000 times, and the distribution of the 10,000 bootstrap means gives us an estimate of the bootstrap sampling distribution.
+
+```{r}
+set.seed(10312022)
+boot_samp_means <- replicate(10000, 
+                             mean(sample(boot_pop, 100,
+                                         replace=TRUE)))
+```
+
+The *bootstrap sampling distribution* looks like the following.
+
+```{r}
+#| warning: false
+#| fig-width: 5
+#| fig-height: 3
+
+boot_sampling_dist <- ggplot(as.data.frame(boot_samp_means), 
+                        aes(x=boot_samp_means, y=..density..)) + 
+      geom_histogram(bins = 45, 
+                 color="black", fill="gray") + 
+      xlab("Average Food Safety Scores for a SRS of 100") +
+      ylab("Density") +
+      ggtitle("Bootstrap Sampling Distribution")
+
+boot_sampling_dist
+```
+
+A few things to note about this bootstrap sampling distribution:
+
+-   The overall shape of the bootstrap sampling distribution mirrors the general shape of the true sampling distribution.
+
+-   The mean of the sampling distribution, `r round(mean(boot_samp_means), 2)`, matches the mean of the bootstrap population, `r round(mean(boot_pop), 2)`, which in turn matches the original sample mean. BUT, it does not match the mean of the original population, `r round(mean(fs_scores), 2)`.
```

Comment:
I wonder the students might come away with the wrong take-home message here. That the mean of the sample is the same as the mean of the bootstrap population: that seems important. That that mean isn't the same as the mean of the original population: that seems useful to note. But the first one - that the mean of the sampling distribution matches the mean of the bootstrap population and thus the original sample - wasn't that just an artifact? If you change the seed, it's not longer true (I suspect).